services/secret-service/src/route/callback/handle-oauth2-password.js

```diff
@@ -0,0 +1,52 @@
+const moment = require('moment');
+const authFlowManager = require('../../auth-flow-manager');
+const AuthClientDAO = require('../../dao/auth-client');
+
+module.exports = async function handleOAuth2Password(data) {
+    const {
+        authClientId, username, password, scope, name,
+    } = data;
+
+    const authClient = await AuthClientDAO.findById(authClientId);
+
+    if (!authClient) {
+        throw new Error('AuthClient not found');
+    }
+
+    const resolvedUsername = username || authClient.username;
+    const resolvedPassword = password || authClient.password;
+    const resolvedScope = scope || authClient.scope;
```

Comment:
I may have missed it, but isn't `authClient.scope` actually `authClient.predefinedScope`?

Reply:
That's right, thanks for catching